File: cpp/june/day_20_Permutation Sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> levels, nums(n+1);
        string res = "";
        int i, j;
        k--;
        for(i = 2; i <= n; i++){
            levels.push_back(k%i);
            k = k / i;
        }
        levels.push_back(0);
        reverse(levels.begin(), levels.end());
        
        for(i = 0; i < n; i++){
            nums[i] = i+1;
        }

        for(i = 1; i < n; i++){
            j = levels[i];
            res += to_string(nums[j]);
            nums.erase(nums.begin() + j);
        }
        res += to_string(nums[0]);
        return res;
        
    }
};

File: python3/june/day_20_Permutation Sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        levels = []
        nums = [None] * (n+1)
        res = ""
        k -= 1
        for i in range(2, n+1):
            levels.append(k%i)
            k = k // i

        levels.append(0)
        levels.reverse()

        for i in range(0, n):
            nums[i] = i+1

        for i in range(1, n):
            j = levels[i]
            res += str(nums[j])
            nums.pop(j)

        res += str(nums[0])
        return res
